target/riscv/translate.c

```diff
@@ -502,12 +502,50 @@ static void gen_jal(DisasContext *ctx, int rd, target_ulong imm)
             return;
         }
     }
-    gen_set_gpr_const(rd, ctx->pc_succ_insn);
+    // For CHERI the result is an offset relative to PCC.base
+    gen_set_gpr_const(rd, ctx->pc_succ_insn - pcc_base(ctx));
 
     gen_goto_tb(ctx, 0, ctx->base.pc_next + imm, /*bounds_check=*/true); /* must use this for safety */
     ctx->base.is_jmp = DISAS_NORETURN;
 }
 
+static void gen_jalr(DisasContext *ctx, int rd, int rs1, target_ulong imm)
```

Comment:
I we change this to
`static void gen_jalr(DisasContext *ctx, int rd, int rs1, target_ulong imm)` we should be able to keep it in the rvi.c file to reduce the diff to upstream?